### PR TITLE
chore: App-router-migration`(/availability`) add pages / extract components into `/module`

### LINKED ISSUE #16293: [umbrella for appDir migration] Add missing routes to App Router for `/availability` route group
## Nested Routes under `/availability` route group:
- [ ] `availability/[schedule].tsx`
- [ ] `availability/index.tsx`
- [ ] `availability/troubleshoot.tsx`

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes #16293
- Fixes CAL-4192, CAL-4193

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Navigate to:

1. `future/availability`
2. `future/availability/[schedule]`
3. `future/availability/troubleshoot`
4. `/availability`
5. `/availability/[schedule]`
6.  `/availability/troubleshoot`

### CODE REVIEW

File: apps/web/modules/availability/[schedule]/schedule-view.tsx [hunk 1]
```diff
@@ -27,8 +28,7 @@ export const AvailabilitySettingsWebWrapper = () => {
     }
   );
 
-  const { data: travelSchedules, isPending: isPendingTravelSchedules } =
-    trpc.viewer.getTravelSchedules.useQuery();
+  const { data: travelSchedules } = trpc.viewer.getTravelSchedules.useQuery();
```
Comment (hbjORbj) [inline apps/web/modules/availability/[schedule]/schedule-view.tsx:31]: `isPending` was not being used anywhere

**Review by sean-brydon — COMMENTED** [review]
Code looks good to me! 

Will test soon

**Review by sean-brydon — DISMISSED** [review]
future/availability :white_check_mark: 
future/availability/[schedule] :white_check_mark: 
future/availability/troubleshoot :white_check_mark: 

/availability :white_check_mark: 
/availability/[schedule] :white_check_mark: 
/availability/troubleshoot :white_check_mark: 


LGTM: I am noticing an warning in console when loading these pages 

``` ⚠ metadata.metadataBase is not set for resolving social open graph or twitter images, using "http://localhost:3000". See https://nextjs.org/docs/app/api-reference/functions/generate-metadata#metadatabase
`themeBasis` is required for booking page theme support. Not providing it will cause theme flicker.
Warning: A props object containing a "key" prop is being spread into JSX:
  let props = {key: someKey, storageKey: ..., forcedTheme: ..., themeSupport: ..., nonce: ..., enableColorScheme: ..., enableSystem: ..., attribute: ..., children: ...};
  <null {...props} />```
  
  I wonder if that is just our page wrapper causing some issues with theme here.

**Comment by hbjORbj** [comment]
Thanks for the review, Sean! I fixed the warning in PR #16496 :D

**Comment by anikdhabal** [comment]
@hbjORbj fix the conflict

**Review by anikdhabal — DISMISSED** [review]
Fix the conflict
Approve on behalf of Sean

File: packages/lib/bookingSuccessRedirect.ts [hunk 2]
```diff
@@ -78,8 +80,8 @@ export const useBookingSuccessRedirect = () => {
       navigateInTopWindow(url.toString());
       return;
     }
-    const newSearchParams = getNewSeachParams({ query });
-    return router.push(`/booking/${booking.uid}?${newSearchParams.toString()}`);
+    const newSearchParams = getNewSeachParams({ query, searchParams: searchParams ?? undefined });
+    return router.push(`/booking/${booking.uid}${isEmbed ? "/embed" : ""}?${newSearchParams.toString()}`);
```
Comment (hariombalhara) [inline packages/lib/bookingSuccessRedirect.ts:84]: Passing all query params as is and ensuring that /embed is present. So, that the appropriate header can be added(either by middleware or by withEmbedSSR)

File: packages/embeds/embed-core/playwright/lib/testUtils.ts [hunk 3]
```diff
@@ -38,12 +38,44 @@ export const getEmbedIframe = async ({
...
+    });
+  };
+
+  page.on("pageerror", (error) => {
+    console.error(`Page error: ${error.message}`);
+  });
+
+  page.on("requestfailed", (request) => {
+    console.error(`Failed request: ${request.url()}, ${request.failure()?.errorText}`);
+  });
```
Comment (hariombalhara) [inline packages/embeds/embed-core/playwright/lib/testUtils.ts:66]: Now the browser console logs would be available inside CI test log to be able to easily identify what was going on in the embed. This should give us a good idea why embed test failed.

**Comment by hariombalhara** [comment]
I accidentally did a force push on one of the commits of this branch.  FYI, in case anyone face any issues pulling

File: packages/embeds/embed-core/playwright/lib/testUtils.ts (hunk 3 above)
Comment (anikdhabal) [reply packages/embeds/embed-core/playwright/lib/testUtils.ts]: Yeah, this makes sense

File: packages/lib/bookingSuccessRedirect.ts [hunk 4]
```diff
@@ -80,7 +80,17 @@ export const useBookingSuccessRedirect = () => {
...
     }
-    const newSearchParams = getNewSeachParams({ query, searchParams: searchParams ?? undefined });
+
+    // TODO: Abstract it out and reuse at other places where we navigate within the embed. Though this is needed only in case of hard navigation happening but we aren't sure where hard navigation happens and where a soft navigation
+    // This is specially true after App Router it seems
+    const headersRelatedSearchParams = searchParams
+      ? {
+          "flag.coep": searchParams.get("flag.coep"),
+        }
+      : {};
```
Comment (anikdhabal) [inline packages/lib/bookingSuccessRedirect.ts:90]: Need to make urlsearchparam instance

**Review by hariombalhara — APPROVED** [review]
Fixed embed issue. Tests passing now.

**Pull request merged**